### PR TITLE
fixes #24504 http logging

### LINKED ISSUE #24504: .NET 6: HTTP Logging
Replace https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/fundamentals/http-logging/samples/6.x with the current template generated code. This article was written with .NET 6 preview 1.

@smandia  do you want this one?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7c772b2a-5d37-84af-624c-9022efe0758b
* Version Independent ID: b39594cc-9d1c-611c-e3e7-8755d74253a6
* Content: [HTTP Logging in .NET Core and ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-logging/?view=aspnetcore-6.0)
* Content Source: [aspnetcore/fundamentals/http-logging/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/http-logging/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @jkotalik
* Microsoft Alias: **jukotali**

### LINKED ISSUE #24538: Capture logs within ASP.NET Core startup code - NO EXAMPLE GIVEN FOR NET CORE 6.0
[Enter feedback here]

Please add the correct code sample.
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7184a308-a254-9350-a5cd-5bced1f369ae
* Version Independent ID: 726e3bf1-f367-d733-8933-bccc04da0e16
* Content: [Logging in .NET Core and ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-6.0)
* Content Source: [aspnetcore/fundamentals/logging/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/logging/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by smandia**
Fixes #24504 Fixes #24538
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/http-logging/?view=aspnetcore-6.0&branch=pr-en-us-24533)

### CODE REVIEW

File: aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
+
+app.UseRouting();
+
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapGet("/", async context =>
+    {
+        await context.Response.WriteAsync("Hello World!");
+    });
+});
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs:41]:
```suggestion
app.MapGet("/", () => "Hello World!");
```

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/fundamentals/http-logging/index.md [hunk 2]
```diff
@@ -35,23 +35,28 @@ HTTP Logging ***can reduce the performance of an app***, especially when logging
 
 HTTP Logging is enabled with `UseHttpLogging`, which adds HTTP logging middleware.
 
-[!code-csharp[](samples/6.x/Startup.cs?name=snippet&highlight=3)]
+[!code-csharp[](samples/6.x/Program.cs?name=snippet&highlight=10)]
 
-By default, HTTP Logging logs common properties such as path, status-code, and headers for requests and responses. The output is logged as a single message at `LogLevel.Information`.
+By default, HTTP Logging logs common properties such as path, status-code, and headers for requests and responses. Please add in appsettings.Development.json file below code to see the logs in putput window.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/index.md:40]:
```suggestion
By default, HTTP Logging logs common properties such as path, status-code, and headers for requests and responses. Add the following line to the *appsettings.Development.json* file so the HTTP logs are displayed.
```

File: aspnetcore/fundamentals/http-logging/index.md [hunk 3]
```diff
@@ -35,23 +35,28 @@ HTTP Logging ***can reduce the performance of an app***, especially when logging
 
 HTTP Logging is enabled with `UseHttpLogging`, which adds HTTP logging middleware.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/index.md:36]:
```suggestion
HTTP Logging is enabled with <xref:Microsoft.AspNetCore.Builder.HttpLoggingBuilderExtensions.UseHttpLogging%2A>, which adds HTTP logging middleware.
```

File: aspnetcore/fundamentals/http-logging/index.md [hunk 4]
```diff
@@ -35,23 +35,28 @@ HTTP Logging ***can reduce the performance of an app***, especially when logging
...
+ ```
+
+The output is logged as a single message at `LogLevel.Information`.
 
 ![Sample request output](_static/requestlog.png)
 
 ## HTTP Logging options
 
-To configure the HTTP logging middleware, call `AddHttpLogging` in `ConfigureServices`.
+To configure the HTTP logging middleware, call `AddHttpLogging` in `Program.cs`.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/index.md:51]:
```suggestion
To configure the HTTP logging middleware, call <xref:Microsoft.Extensions.DependencyInjection.HttpLoggingServicesExtensions.AddHttpLogging%2A> in `Program.cs`.
```

File: aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs [hunk 5]
```diff
@@ -1,26 +1,38 @@
...
-using Microsoft.Extensions.Logging;
-
-namespace HttpLoggingSample
+using Microsoft.AspNetCore.HttpLogging;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+//builder.Services.AddRazorPages();
+#region Addservices
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs:7]:
```suggestion
#region Addservices
using Microsoft.AspNetCore.HttpLogging;

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.
//builder.Services.AddRazorPages();
```

File: aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs [hunk 6]
```diff
@@ -1,26 +1,38 @@
...
+    logging.RequestHeaders.Add("My-Request-Header");
+    logging.ResponseHeaders.Add("My-Response-Header");
+    logging.MediaTypeOptions.AddText("application/javascript");
+    logging.RequestBodyLogLimit = 4096;
+    logging.ResponseBodyLogLimit = 4096;
+
+});
+#endregion
+
+#region snippet
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs:21]: Move `#region snippet` to the top, show entire file. Use highlighting.

File: aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs (hunk 5 above)
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs:7]: Use highlighting.

File: aspnetcore/fundamentals/http-logging/index.md [hunk 7]
```diff
@@ -79,19 +84,19 @@ To configure the HTTP logging middleware, call `AddHttpLogging` in `ConfigureSer
...
 
-[!code-csharp[](samples/6.x/Startup.cs?name=configureservices&highlight=7)]
+[!code-csharp[](samples/6.x/Program.cs?name=Addservices&highlight=5)]
 
 ### `ResponseHeaders`
 
 `ResponseHeaders` are a set of HTTP Response Headers that are allowed to be logged. Header values are only logged for header names that are in this collection.
 
-[!code-csharp[](samples/6.x/Startup.cs?name=configureservices&highlight=8)]
+[!code-csharp[](samples/6.x/Program.cs?name=Addservices&highlight=6)]
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/index.md:93]: The first time you show `builder.Services.AddHttpLogging`, show the entire file. The following times just the snippet you're now using.

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/fundamentals/http-logging/index.md (hunk 7 above)
Comment (smandia) [reply aspnetcore/fundamentals/http-logging/index.md:93]: done in line number 53

File: aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs (hunk 6 above)
Comment (smandia) [reply aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs:21]: Is it still needed. since nested regions . It is a curious question.

**Comment by smandia** [comment]
@Rick-Anderson  Please review again . Apologies for the delay in making the changes.

**Comment by smandia** [comment]
@Rick-Anderson  Please share your views

**Comment by Rick-Anderson** [comment]
@smandia  we're off today for MLK day. I'll review tomorrow.

File: aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs [hunk 8]
```diff
@@ -1,26 +1,38 @@
...
-using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
-using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
-
-namespace HttpLoggingSample
+#region snippet
+#region Addservices
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/samples/6.x/Program.cs:2]:
```suggestion
#region snippet_Addservices
```

File: aspnetcore/fundamentals/http-logging/index.md [hunk 9]
```diff
@@ -33,25 +33,30 @@ HTTP Logging ***can reduce the performance of an app***, especially when logging
...
 
 ![Sample request output](_static/requestlog.png)
 
 ## HTTP Logging options
 
-To configure the HTTP logging middleware, call `AddHttpLogging` in `ConfigureServices`.
+To configure the HTTP logging middleware, call <xref:Microsoft.Extensions.DependencyInjection.HttpLoggingServicesExtensions.AddHttpLogging%2A> in `Program.cs`.
 
-[!code-csharp[](samples/6.x/Startup.cs?name=configureservices)]
+[!code-csharp[](samples/6.x/Program.cs)]
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/index.md:53]:
```suggestion
[!code-csharp[](samples/6.x/Program.cs?name=snippet_Addservices)]
```

File: aspnetcore/fundamentals/http-logging/index.md [hunk 10]
```diff
@@ -33,25 +33,30 @@ HTTP Logging ***can reduce the performance of an app***, especially when logging
...
 
-[!code-csharp[](samples/6.x/Startup.cs?name=configureservices)]
+[!code-csharp[](samples/6.x/Program.cs?name=snippet_Addservices)]
 
 ### `LoggingFields`
 
 `HttpLoggingOptions.LoggingFields` is an enum flag that configures specific parts of the request and response to log. `LoggingFields` defaults to `RequestPropertiesAndHeaders | ResponsePropertiesAndHeaders`.
 
-[!code-csharp[](samples/6.x/Startup.cs?name=configureservices&highlight=6)]
+[!code-csharp[](samples/6.x/Program.cs?name=Addservices&highlight=4)]
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/index.md:59]: [!code-csharp[](samples/6.x/Program.cs?name=snippet_Addservices)]

Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/index.md:59]:
```suggestion
[!code-csharp[](samples/6.x/Program.cs?name=snippet_Addservices)]
```

File: aspnetcore/fundamentals/http-logging/index.md [hunk 11]
```diff
@@ -33,25 +33,30 @@ HTTP Logging ***can reduce the performance of an app***, especially when logging
 
 ## Enabling HTTP logging
 
-HTTP Logging is enabled with `UseHttpLogging`, which adds HTTP logging middleware.
+HTTP Logging is enabled with <xref:Microsoft.AspNetCore.Builder.HttpLoggingBuilderExtensions.UseHttpLogging%2A>, which adds HTTP logging middleware.
 
-[!code-csharp[](samples/6.x/Startup.cs?name=snippet&highlight=3)]
+[!code-csharp[](samples/6.x/Program.cs?name=snippet&highlight=10)]
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/http-logging/index.md:38]:
```suggestion
[!code-csharp[](samples/6.x/Program.cs?name=snippet_Addservices)]
```

**Pull request closed without merging**